### PR TITLE
List: Add Sizing to List

### PR DESCRIPTION
**Pull request by rlingineni**
Add multiple sizes for list sizing.

### Summary
We support the 'Text' Element in List, but not different font sizes. This adds more font sizing support to List to align with `Text` and passes it to the rest of the children.

Before:
image

AFter:
image

#### What changed?

Adding support for a `size` prop to for varying and uniform font-sizes

#### Why?
List is used alongside paragraphs and other sets of text. Since it is primarily a text element, aligning with a text sizing here seems good.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "css.lint.validProperties": ["composes"],
   "editor.codeActionsOnSave": {
+    "source.organizeImports": false,
```
Comment (diyorbek) [inline .vscode/settings.json:4]: what's the motivation for this change?

Comment (rlingineni) [reply .vscode/settings.json:4]: VSCode's sorting and es-lint autofix was was breaking my changes. Every time on auto-save, vscode would override the eslint auto-fix to sort imports.

So I added this to disable the vscode import organization.

Comment (diyorbek) [reply .vscode/settings.json:4]: So is this a temporary change then? Without this rule we won't have sorted imports for the future changes. Also, I think i didn't understand how exactly import sorting is breaking your changes. Could you elaborate more?

Comment (rlingineni) [reply .vscode/settings.json:4]: Nope, this is a permanent fix. Sorry, let me clarify.

We have an eslint rule called [sort-imports](https://eslint.org/docs/latest/rules/sort-imports). On save, vscode automatically runs `source.fixAll.eslint` so it sorts the order, and fixes problem.

VSCode, also ships a feature with `source.organizeImports`. When I save my file after auto-sorting my imports. VSCode overrides that before saving, and uses its own formatting.

Correct:
image

Becomes this after save:
image

If I don't make this change here, I can't ship my code, because my editor is broken otherwise.

Hopefully that clarifies

Comment (diyorbek) [reply .vscode/settings.json:4]: Oh, now I see it. Thanks for clarifying 👍

**Review by diyorbek — APPROVED** [review]
